subprojects/gradle/runner/service/src/main/kotlin/com/avito/runner/service/IntentionExecutionServiceImplementation.kt

```diff
@@ -45,36 +48,39 @@ class IntentionExecutionServiceImplementation(
 
     @OptIn(ExperimentalCoroutinesApi::class)
     override fun start(scope: CoroutineScope): IntentionExecutionService.Communication {
-        scope.launch {
-            launch {
+        scope.launch(CoroutineName("intention-execution-service")) {
+            launch(CoroutineName("device-workers")) {
                 while (!devices.isClosedForReceive) {
-                    val device = devices.receive()
-                    DeviceWorker(
-                        intentionsRouter = intentionsRouter,
-                        device = device,
-                        outputDirectory = outputDirectory,
-                        deviceListener = CompositeDeviceListener(
-                            listOf(
-                                MessagesDeviceListener(messages),
-                                DeviceLogListener(device.logger),
-                                deviceMetricsListener
-                            )
-                        ),
-                        testListener = testListener,
-                        dispatchers = deviceWorkersDispatcher,
-                        timeProvider = timeProvider
-                    ).run(scope)
+                    val device = devices.receiveOrNull()
```

Comment:
when we suspend on a channel that was closed it throws an exception on `receive`.
The opposite is used `receiveOrNull` to simply get `null`